modules/drivers/druid-jdbc/src/metabase/driver/druid_jdbc.clj

```diff
@@ -0,0 +1,183 @@
+
+(defmethod sql-jdbc.sync/database-type->base-type :druid-jdbc
+  [_ database-type]
+  ({:TIMESTAMP            :type/DateTime
+    :VARCHAR              :type/Text
+    :DECIMAL              :type/Decimal
+    :FLOAT                :type/Float
+    :REAL                 :type/Float
+    :DOUBLE               :type/Float
+    :BIGINT               :type/BigInteger
+    :COMPLEX<json>        :type/JSON
+    :COMPLEX<hyperUnique> :type/DruidHyperUnique}
+   database-type
+   :type/*))
+
+(defmethod sql-jdbc.execute/read-column-thunk [:druid-jdbc Types/TIMESTAMP]
+  [_ ^ResultSet rs _ ^long i]
+  (fn []
+    (t/instant (.getObject rs i))))
+
+;; Druid's COMPLEX<...> types are encoded as JDBC's other -- 1111. Values are rendered as string.
+(defmethod sql-jdbc.execute/read-column-thunk [:druid-jdbc 1111]
+  [_ ^ResultSet rs _ ^long i]
+  (fn []
+    (let [o (.getObject rs i)]
+      (cond-> o
+        (string? o) (str/replace  #"^\"|\"$" "")))))
+
+(defn- date-trunc [unit expr] [:date_trunc (h2x/literal unit) expr])
+
+(defmethod sql.qp/date [:druid-jdbc :default] [_ _ expr] expr)
+(defmethod sql.qp/date [:druid-jdbc :minute] [_ _ expr] (date-trunc :minute expr))
+(defmethod sql.qp/date [:druid-jdbc :hour] [_ _ expr] (date-trunc :hour expr))
+(defmethod sql.qp/date [:druid-jdbc :day] [_ _ expr] (date-trunc :day expr))
+(defmethod sql.qp/date [:druid-jdbc :week] [_ _ expr] (date-trunc :week expr))
+(defmethod sql.qp/date [:druid-jdbc :month] [_ _ expr] (date-trunc :month expr))
+(defmethod sql.qp/date [:druid-jdbc :quarter] [_ _ expr] (date-trunc :quarter expr))
+(defmethod sql.qp/date [:druid-jdbc :year] [_ _ expr] (date-trunc :year expr))
+
+(defn- time-extract [unit expr] [:time_extract expr (h2x/literal unit)])
+
+(defmethod sql.qp/date [:druid-jdbc :minute-of-hour] [_ _ expr] (time-extract :minute expr))
+(defmethod sql.qp/date [:druid-jdbc :hour-of-day] [_ _ expr] (time-extract :hour expr))
+(defmethod sql.qp/date [:druid-jdbc :day-of-week] [_ _ expr] (time-extract :dow expr))
+(defmethod sql.qp/date [:druid-jdbc :day-of-month] [_ _ expr] (time-extract :day expr))
+(defmethod sql.qp/date [:druid-jdbc :day-of-year] [_ _ expr] (time-extract :doy expr))
+(defmethod sql.qp/date [:druid-jdbc :week-of-year] [_ _ expr] (time-extract :week expr))
+(defmethod sql.qp/date [:druid-jdbc :month-of-year] [_ _ expr] (time-extract :month expr))
+(defmethod sql.qp/date [:druid-jdbc :quarter-of-year] [_ _ expr] (time-extract :quarter expr))
+
+(defmethod sql.qp/current-datetime-honeysql-form :druid-jdbc
+  [_]
+  [:raw "CURRENT_TIMESTAMP"])
+
+(defmethod sql.qp/add-interval-honeysql-form :druid-jdbc
+  [_ hsql-form amount unit]
+  [:TIMESTAMPADD (h2x/identifier :type-name unit) (h2x/->integer amount) hsql-form])
+
+;; Query used for field values computation is not compatible with COMPLEX<...> types.
+(defmethod driver/field-values-compatible? :druid-jdbc
+  [_driver {:keys [database_type] :as _field}]
+  (not (re-find #"COMPLEX<" (str database_type))))
```

Comment:
You should probably just use `str/includes?` here since this regex isn't doing anything regexy

Reply:
Thanks for the catch. I've removed the method completely in favor of `:type/field-values-unsupported`.